cape-char.el

```diff
@@ -28,41 +28,52 @@
 
 (autoload 'thing-at-point-looking-at "thingatpt")
 
+(defmacro cape-char--ensure-str (char-or-str)
+  "IF-expression to convert CHAR-OR-STR to string only if it's a char"
+  `(if (characterp ,char-or-str)
+      (char-to-string ,char-or-str) ,char-or-str))
+
+(defmacro cape-char--ensure-char (char-or-str)
+  "IF-expression to convert CHAR-OR-STR to char only if it's a string"
+  `(if (stringp ,char-or-str)
+      (string-to-char ,char-or-str) ,char-or-str))
```

Comment:
I would prefer if these very short functions are inlined. They are only used once and twice respectively.

Reply:
Indeed they are very short, but since they're also needed in the translation hash function, the number of references would be 2 and 3 respectively. See while I was writing the code I found having to do `if (stringp (char-to-string ...) ...` verbose and repetitive. So I extracted it to functions to make the code more concise (easier to read), and the changes to `cape-char--define` minimal.

But sure I can replace all references for these to the `if` full form if you insist. Otherwise since we're going to remove `eval-when-compile` I guess they should stay as functions rather than macros to be more useful as public functions.

Reply:
Continued at: [comment after these review comments](https://github.com/minad/cape/pull/91#issuecomment-1715190400)

Suggestion applied in: [1e89ad2](https://github.com/minad/cape/pull/91/commits/1e89ad26ae0a8ce72a7339efe542224d6e38cfb6)